migrator/migrator.go

```diff
@@ -0,0 +1,269 @@
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Second*20)
+	defer cancel()
+
+	nodeNames, err := getMasterNodes(m.k8sClient, m.masterNodeLabel)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	memberListResponse, err := m.etcdClient.Cluster.MemberList(ctxWithTimeout)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	memberCount := len(memberListResponse.Members)
+
+	fmt.Printf("Found %d etcd members in the cluster.\n", memberCount)
+
+	if memberCount == masterNodeCount {
+		fmt.Printf("Etcd cluster already has 3 nodes. Nothing to do. Exiting.\n")
+		return nil
+	} else if memberCount == 2 {
+		// continue migration that was interrupted in the middle of the process
+		err = m.addNodeToEtcdCluster(ctx, nodeNames, 3)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+	} else if memberCount == 1 {
+		//  ensure that first node has proper etcd peer url set to etcd1.xxxx.xxxx.xxx
+		err = m.fixFirstNodePeerUrl(ctx, memberListResponse.Members)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		// add second node to the etcd cluster
+		err = m.addNodeToEtcdCluster(ctx, nodeNames, 2)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// add third node to the etcd cluster
+		err = m.addNodeToEtcdCluster(ctx, nodeNames, 3)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	} else {
+		fmt.Printf("unexpected number of nodes in etcd cluster\n")
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("found %d nodes in etcd cluster", memberCount))
+	}
+
+	fmt.Printf("ETCD cluster migration succesfuly finished.\n\n")
+	fmt.Printf("Sleeping forever.\n")
+	select {}
+}
+
+// fixFirstNodePeerUrl ensure the peerURL for the first node in etcdcluster is properly set
+// as it can have 'localhost' value from the previous version fo k8scloudconfig.
+func (m *Migrator) fixFirstNodePeerUrl(ctx context.Context, etcdMembers []*etcdserver.Member) error {
+	id := etcdMembers[0].ID
+	peerUrls := []string{etcdPeerName(m.etcdStartingIndex, m.baseDomain)}
+
+	_, err := m.etcdClient.Cluster.MemberUpdate(ctx, id, peerUrls)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	fmt.Printf("Updated first node PeerUrls to %s.\n", peerUrls)
+	return nil
+}
+
+// addNodeToEtcdCluster configure etcd3 service on the second or third node in order
+// to join the existing cluster via k8s job executed on the node and after that
+// it will add the node to the etcd cluster via etcdv3 client API.
+
+func (m *Migrator) addNodeToEtcdCluster(ctx context.Context, nodeNames []string, nodeCount int) error {
+	// nodeCount can only be 2 or 3
+	// 2  when adding second node to a single node etcd cluster
+	// 3 when adding third node to two node etcd cluster
+	if nodeCount != 2 && nodeCount != 3 {
+		return microerror.Maskf(executionFailedError, "nodeCount can only have values 2 or 3")
+	}
+
+	// execute commands on the node so the node in order to configure new etcd3 member to join existing cluster
+	{
+		nodeName := nodeNames[nodeCount-1]
```

Comment:
This here panics when called the wrong way. Some error handling might make sense. Uncertain.

Reply:
the two `if`  above should sort any wrong calls